# DataViz Project - Airplane Crashes Since 1908

## Exploratory Data Analysis

### 1. Reading the data

Read and convert the data into Python dictionaries

In [18]:
import csv 

file_path = "dataset/clean_v1_Airplane_Crashes_and_Fatalities_Since_1908.csv"

with open(file_path) as file:
    reader = csv.DictReader(file) #row iterator
    crashes = list(reader)
    
crashes

[OrderedDict([('Date', '09/17/1908'),
              ('Time', '17:18'),
              ('Location', 'Fort Myer, Virginia'),
              ('Latitude', '38.88243103'),
              ('Longitude', '-77.08074951'),
              ('Operator', 'Military - U.S. Army'),
              ('Type', 'Wright Flyer III'),
              ('Aboard', '2'),
              ('Fatalities', '1'),
              ('Ground', '0'),
              ('Summary',
               'During a demonstration flight, a U.S. Army flyer flown by Orville Wright nose-dived into the ground from a height of approximately 75 feet, killing Lt. Thomas E. Selfridge who was a passenger. This was the first recorded airplane fatality in history.  One of two propellers separated in flight, tearing loose the wires bracing the rudder and causing the loss of control of the aircraft.  Orville Wright suffered broken ribs, pelvis and a leg.  Selfridge suffered a crushed skull and died a short time later.')]),
 OrderedDict([('Date', '07/12/1912'),
    

Put the data into a Pandas dataframe:



In [17]:
import pandas as pd

dframe = pd.DataFrame(crashes)
dframe



Date   Time                                           Location  \
0     09/17/1908  17:18                                Fort Myer, Virginia   
1     07/12/1912  06:30                          Atlantic City, New Jersey   
2     08/06/1913                        Victoria, British Columbia, Canada   
3     09/09/1913  18:30                                 Over the North Sea   
4     10/17/1913  10:30                         Near Johannisthal, Germany   
5     03/05/1915  01:00                                    Tienen, Belgium   
6     09/03/1915  15:20                              Off Cuxhaven, Germany   
7     07/28/1916                                     Near Jambol, Bulgeria   
8     09/24/1916  01:00                                Billericay, England   
9     10/01/1916  23:45                               Potters Bar, England   
10    11/21/1916                                            Mainz, Germany   
11    11/28/1916  23:45                       Off West Hartlepool, England   
12    03/04/1917                                        Near Gent, Belgium   
13    03/30/1917                                      Off Northern Germany   
14    05/14/1917  05:15                       Near Texel Island, North Sea   
15    06/14/1917  08:45                     Off Vlieland Island, North Sea   
16    08/21/1917  07:00                                Off western Denmark   
17    10/20/1917  07:45                             Near Luneville, France   
18    04/07/1918  21:30                             Over the Mediterranean   
19    05/10/1918                             Off Helgoland Island, Germany   
20    08/11/1918  10:00                          Ameland Island, North Sea   
21    12/16/1918                                     Elizabeth, New Jersey   
22    05/25/1919                                           Cleveland, Ohio   
23    07/19/1919                                     Dix Run, Pennsylvania   
24    10/02/1919                                        Newcastle, England   
25    10/14/1919                                    Cantonsville, Maryland   
26    10/20/1919                                           English Channel   
27    10/30/1919                                   Long Valley, New Jersey   
28    03/10/1920                                        New Paris, Indiana   
29    03/30/1920                                        Newark, New Jersey   
...          ...    ...                                                ...   
5238  11/13/2008  11:35                                     Fallujah, Iraq   
5239  11/16/2008  10:32         Thormanby Island, British Columbia, Canada   
5240  11/27/2008  17:00                              Off Perpignan, France   
5241  12/03/2008  12:13                              San Juan, Puerto Rico   
5242  12/11/2008  06:25                             Off Sabine Pass, Texas   
5243  12/15/2008  16:40                       Off Turks and Caicos Islands   
5244  01/04/2009  15:30                               Near Houma Louisiana   
5245  01/15/2009  15:06                                 New York, New York   
5246  02/07/2009  14:00                                 Manacapuru, Brazil   
5247  02/07/2009  06:30                                    Trigoria, Italy   
5248  02/12/2009  22:17                          Clarence Center, New York   
5249  02/15/2009  16:05                                      Chanco, Chile   
5250  02/20/2009  04:10                                       Luxor, Egypt   
5251  02/25/2009  10:31                             Amsterdam, Netherlands   
5252  03/09/2009  05:40                              Lake Victoria, Uganda   
5253  03/12/2009  09:48                        Off St. Johns, Newfoundland   
5254  03/22/2009  14:28                                     Butte, Montana   
5255  03/23/2009  06:48                                       Tokyo, Japan   
5256  04/01/2009  14:00                              Off Crimond, Scotland   
5257  04/06/2009  12:30           

Alternatively, the two previous blocks can be simply handled by Pandas method "read_csv", as below:

In [21]:
import pandas as pd

dframe = pd.read_csv(file_path)

# get information about data entries by column:
dframe.info()
# get a brief statistical summary of the data
dframe.describe()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5268 entries, 0 to 5267
Data columns (total 11 columns):
Date          5268 non-null object
Time          3049 non-null object
Location      5248 non-null object
Latitude      5240 non-null float64
Longitude     5240 non-null float64
Operator      5250 non-null object
Type          5241 non-null object
Aboard        5246 non-null float64
Fatalities    5256 non-null float64
Ground        5246 non-null float64
Summary       4878 non-null object
dtypes: float64(5), object(6)
memory usage: 452.8+ KB


Latitude    Longitude       Aboard   Fatalities       Ground
count  5240.000000  5240.000000  5246.000000  5256.000000  5246.000000
mean     27.010367   -16.061076    27.554518    20.068303     1.608845
std      24.570333    83.419121    43.076711    33.199952    53.987827
min     -77.529716  -176.669861     0.000000     0.000000     0.000000
25%      10.944428   -82.810095     5.000000     3.000000     0.000000
50%      34.392923    -9.149480    13.000000     9.000000     0.000000
75%      43.801488    37.615021    30.000000    23.000000     0.000000
max      80.449997   178.800476   644.000000   583.000000  2750.000000

From the info() method we can see there are several missing values - there are 5268 entries

Date          5268 
Time          3049  *-> missing 2219 values*
Location      5248  *-> missing 20 values*
Latitude      5240  *-> missing 28 values (generated from Location, missing 8)*
Longitude     5240  *-> missing 28 values (generated from Location, missing 8)*
Operator      5250  *-> missing 18 values*
Type          5241  *-> missing 19 values*
Aboard        5246  *-> missing 22 values*
Fatalities    5256  *-> missing 12 values*
Ground        5246  *-> missing 22 values*
Summary       4878  *-> missing 390 values*